archive/frames/filters.py

```diff
@@ -67,8 +72,8 @@ def public_filter(self, queryset, name, value):
                 user_proposals = self.request.user.profile.proposals
                 if user_proposals:
                     return queryset.filter(proposal_id__in=user_proposals)
-
-            return queryset.exclude(public_date__lt=timezone.now())
+            else:
+                return queryset.exclude(public_date__lt=timezone.now())
```

Comment:
FYI, at this point I think `queryset` is already filtered based on the auth info https://github.com/observatorycontrolsystem/science-archive/blob/62ac5e56d5c1c44ab8b98ca6f1514613bb450d6f/archive/frames/views.py#L60

I wonder if this could be simplified further by only having:

```python
if not value:
   return queryset.exclude(public_date__lt=timezone.now())

return queryset
```

Reply:
On second look, you'd also need to check the auth here too. Maybe all this doesn't matter and the duplicate clause gets optimized out by the ORM or Postgres.

Reply:
Yeah was just gonna say - we need to make sure we still include all frames belonging to a user's proposal, public or not, no matter what.